### DISCUSSION TITLE
EMQX broker configuration

### DISCUSSION THREAD

**Discussion by naikvidish** [emqx/emqx#8350] [Q&A] [answered]
Hi,

I want to set a few configurations but not use ENV variables or the config file. Can you tell me at what point does EMQX read these configs so that I can modify these through my code before they are read by EMQX? 

Thanks in advance!

**Comment by zmstone** [comment, marked as answer]
The conf files are read before the service starts

**Reply by naikvidish** [reply]
Thanks for the response!

**Comment by MikeDombo** [comment]
@zmstone Is there a way in EMQX 5 to programmatically update the configuration? 

Or, for example, does EMQX 5 read the [override files](https://www.emqx.io/docs/en/v5.0/admin/cfg.html#layered) and apply their configuration when there are changes?
